File: app/src/main/java/com/x7/steambadger/application/SteamBadgeR.java
package com.x7.steambadger.application;

import android.app.Application;

import com.squareup.picasso.OkHttpDownloader;
import com.squareup.picasso.Picasso;

public class SteamBadgeR extends Application {

	private static SteamBadgeR singleton;

	public SteamBadgeR() {
	}

	@Override
	public void onCreate() {
		super.onCreate();
		SteamBadgeR.singleton = this;

		Picasso.Builder builder = new Picasso.Builder(this);
		builder.downloader(new OkHttpDownloader(this, Integer.MAX_VALUE));
		Picasso built = builder.build();
		Picasso.setSingletonInstance(built);
	}

	public static SteamBadgeR getInstance() {
		return SteamBadgeR.singleton;
	}

}

File: app/src/main/java/com/dougmaitelli/steambadger/application/SteamBadgeR.kt
package com.dougmaitelli.steambadger.application

import androidx.multidex.MultiDexApplication

import com.squareup.picasso.OkHttpDownloader
import com.squareup.picasso.Picasso

class SteamBadgeR : MultiDexApplication() {

    override fun onCreate() {
        super.onCreate()

        val builder = Picasso.Builder(this)
        builder.downloader(OkHttpDownloader(this, Long.MAX_VALUE))
        val built = builder.build()
        Picasso.setSingletonInstance(built)
    }

    init {
        instance = this
    }

    companion object {
        lateinit var instance: SteamBadgeR
    }

    class Event private constructor() {

        companion object {
            const val LOGIN = "login"
            const val LOGOUT = "logout"
        }

    }

    class Param private constructor() {

        companion object {
            const val CUSTOM_URL = "customUrl"
            const val STEAM_ID = "steamId"
        }

    }

}
